eZ/Bundle/EzPublishCoreBundle/Command/CleanupVersionsCommand.php

```diff
@@ -145,15 +146,13 @@ protected function execute(InputInterface $input, OutputInterface $output)
                     $versionsCount
                 ), Output::VERBOSITY_VERBOSE);
 
-                $versions = array_filter($versions, function ($version) use ($removeAll, $removeDrafts, $removeArchived) {
-                    if (
-                        ($removeAll && $version->status !== VersionInfo::STATUS_PUBLISHED) ||
-                        ($removeDrafts && $version->status === VersionInfo::STATUS_DRAFT) ||
-                        ($removeArchived && $version->status === VersionInfo::STATUS_ARCHIVED)
-                    ) {
-                        return $version;
-                    }
-                });
+                if ($removeAll) {
+                    $versions = array_filter($versions, static function ($version) {
+                        if ($version->status !== VersionInfo::STATUS_PUBLISHED) {
```

Comment:
Think we can do just a `return $version->status !== VersionInfo::STATUS_PUBLISHED;` here afaik, right?
As array_filter seems to take a true/false return value, and not he value itself (so this works by accident here): https://php.net/array_filter

Reply:
Good catch, thank you! Fixed in: https://github.com/ezsystems/ezpublish-kernel/pull/2692/commits/8b5dd39c4e4ab50082e4621c6764a9e256aea477